### ISSUE TITLE
Not work export LD_LIBRARY_PATH from nvidia-xinitrc

### ISSUE THREAD

**Issue by Pro-pra** [Witko/nvidia-xrun#27]
I have nvidia lib installed on /usr/lib/nvidia
export LD_LIBRARY_PATH from nvidia-xinitrc not work for me, and GLX extension not initialized.

I resolve this problem:
Add file /etc/ld.so.conf.d/nvidia-xrun.conf 
with path to nvidia libs /usr/lib/nvidia
nvidia-xrun work fine, but on normal mode glx extension for intel stopped working.

Maybe there are other methods to set LD_LIBRARY_PATH not from nvidia-xinitrc?

**Comment by Pro-pra** [comment]
My updated resolve:
in nvidia-xrun add command: 
sudo ldconfig /usr/lib/nvidia at the beginning
sudo ldconfig at the end.

**Comment by Witko** [comment]
Hi @Pro-pra ,
maybe the solution would be to set the LD_LIBRARY_PATH in $HOME/.nvidia-xinitrc. But it seems at this point of time it would be overriden, but if i move the export above the sourcing of this file it should be ok.

**Comment by Pro-pra** [comment]
> adding custom config script in /etc/X11/xinit/nvidia-xinitrc.d/

not work for me, 
`sudo ldconfig /usr/lib/nvidia` work fine.

**Comment by Witko** [comment]
Hi @Pro-pra ,
did you try with new sources? There is no release out yet.

**Comment by Pro-pra** [comment]
yes, i manual move string "export LD_LIBRARY..." to the new location. 
And add script to directory /etc/X11/xinit/nvidia-xinitrc.d/nvidia-xrun.sh

**Comment by Witko** [comment]
Hmm, that's bad. Do you know which library causes the issues?

**Comment by Pro-pra** [comment]
I dont know. In simple scripts to run the program (not for nvidia-xrun) I have everything working LD_LIBRARY_PATH=

**Comment by dpacmittal** [comment]
I have this same issue and thank god someone created this, I was about to go mental just staring that Xorg logs. It seems like once a particular library is loaded, the other one stops working.

I have Intel running KDE Plasma on TTY7. It works fine. However, when I run nvidia-xrun on TTY1 and run glxinfo inside it, it gives me "GLX extensions missing" error. `sudo ldconfig /usr/lib/nvidia-390` on top of nvidia-xrun fixes it. However now if I switch to TTY7 again, Intel starts throwing error.

Timeline:
- Boot.
- KDE Plasma boots on TTY7 on Intel. glxinfo works.
- Press Ctrl + Alt + F1. Login. Run nvidia-xrun (openbox-session)
- Inside openbox glxinfo doesn't work.
- Modify nvidia-xrun and add `ldconfig /usr/lib/nvidia-390` on top. glxinfo inside openbox works now.
- Keep nvidia-xrun running and switch to TTY7, run glxinfo it throws an error.

So basically only one gpu works at a time, it can be either Nvidia or Intel, not both. Is there any better solution to this because this way nvidia-xrun doesn't give any additional benefit over bumblebee.
